kombu/transport/SQS.py

```diff
@@ -153,7 +153,9 @@ def _reset_cycle(self):
     def entity_name(self, name, table=CHARS_REPLACE_TABLE):
         """Format AMQP queue name into a legal SQS queue name."""
         if name.endswith('.fifo'):
-            partial = name.rstrip('.fifo')
+            # SQS allows no dots in queue names except FIFO queues.
+            # For FIFO queues only one dot is allowed in '.fifo' suffix
+            partial = name.split('.')[0]
```

Comment:
Why can't we use 
`partial = name.rsplit('.fifo')[0]`? 
Splitting on dot seems a bit scary. 
Or even `partial = name.split('.fifo')[0]`

Please note **split** instead of **strip**

Reply:
And moreover note that they are handling the **no dots in queue names** by the `CHARS_REPLACE_TABLE` setting. This would just violate DRY principle and bring in scary changes.

Reply:
Why can't you simply do:
```py
partial = name[:-(len('.fifo')]
```

Reply:
Yeah. This is the better method. Slicing is better than splitting. 👍

Reply:
according to https://doeidoei.wordpress.com/2011/10/18/python-slicing-vs-splitting-vs-endswith/